flink-core/src/main/java/org/apache/flink/api/java/typeutils/runtime/RowSerializer.java

```diff
@@ -367,36 +367,42 @@ public int getVersion() {
 	/**
 	 * A {@link TypeSerializerSnapshot} for RowSerializer.
 	 */
-	// TODO not fully functional yet due to FLINK-17520
 	public static final class RowSerializerSnapshot extends CompositeTypeSerializerSnapshot<Row, RowSerializer> {
 
 		private static final int VERSION = 3;
 
-		private static final int VERSION_WITHOUT_ROW_KIND = 2;
+		private static final int LAST_VERSION_WITHOUT_ROW_KIND = 2;
 
-		private boolean legacyModeEnabled = false;
+		private int version = VERSION;
```

Comment:
Maybe rename this to `readVersion`, to better convey its difference with the static `VERSION`.